src/Behat/Gherkin/Node/FeatureNode.php

```diff
@@ -78,6 +80,11 @@ public function __construct(
         $file,
         $line
     ) {
+        // Verify that the feature file is an absolute path.
+        $filesystem = new Filesystem();
+        if (!empty($file) && !$filesystem->isAbsolutePath($file)) {
+            throw new \InvalidArgumentException('The file should be an absolute path.');
```

Comment:
When running from PHPStorm, I hit this exception having `features/api/bien/bienOption.feature` as `$file` value.

Here is my `behat.yml`:

```
default:
    suites:
        default:
            paths:
                - "%paths.base%/features"
            contexts:
                - PropelContext:
                - AppBundle\Utils\Behat\Context\HttpContext:
                    client: '@AppBundle\Utils\Behat\Client'
                - AppBundle\Utils\Behat\Context\ClockContext:
                    clock: '@AppBundle\Utils\Behat\MockClock'
                - ApiContext:
                    tokenStorage: '@security.token_storage'
                    userManager: '@app.user_manager'
                    clientManager: '@fos_oauth_server.client_manager.default'
                    oauth2Storage: '@fos_oauth_server.storage'
                    client: '@AppBundle\Utils\Behat\Client'
                - CoverageContext:
                - FixtureContext:
                    venteManager: '@app.vente_manager'
                    locationManager: '@app.location_manager'
                    emailManager: '@app.email_manager'
                    reportManager: '@app.report_manager'
                    userManager: '@app.user_manager'
                    clientManager: '@fos_oauth_server.client_manager'
                    mediaManager: '@app.media_manager'
                    oAuthStorage: '@app.oauth.storage'
                - PdfContext:
                    client: '@AppBundle\Utils\Behat\Client'
                - UploadContext:
                - ImageContext:
                    client: '@AppBundle\Utils\Behat\Client'
                    imagine: '@Imagine\Image\ImagineInterface'
                - PrevisiteContext:
                    mock: '@Noty\VisiteVirtuelle\Previsite\HttpClient\Plugin\Mock'
                - Noty\Vente\Behat\Context\VenteContext:
                    validator: '@validator'
                    propertyAccessor: '@property_accessor'
                - DiffusionContext:
                    normalizerFactory: '@app.passerelle.normalizer_factory'
                    passerelleManager: '@app.passerelle_manager'
    extensions:
        Caciobanu\Behat\DeprecationExtension:
        Behat\Symfony2Extension:
            kernel:
                env: test
```

the command line launched by PhpStorm:

```
/usr/local/Cellar/php/7.2.7/bin/php /private/var/folders/xy/sry34mvs6l7g7jfrjx64nh7m0000gn/T/ide-behat.php --format PhpStormBehatFormatter --no-interaction --config /Users/fabien/Projets/soqrate/noty/api/behat.yml /Users/fabien/Projets/soqrate/noty/api/features/api/bien/bienOption.feature:36
```

And the PhpStorm output:
```
Testing started at 14:47 ...
/usr/local/Cellar/php/7.2.7/bin/php /private/var/folders/xy/sry34mvs6l7g7jfrjx64nh7m0000gn/T/ide-behat.php --format PhpStormBehatFormatter --no-interaction --config /Users/fabien/Projets/soqrate/noty/api/behat.yml /Users/fabien/Projets/soqrate/noty/api/features/api/bien/bienOption.feature:36

In FeatureNode.php line 86:
                                        
  The file should be an absolute path.  
                                        

behat [-s|--suite SUITE] [-f|--format FORMAT] [-o|--out OUT] [--format-settings FORMAT-SETTINGS] [--init] [--lang LANG] [--name NAME] [--tags TAGS] [--role ROLE] [--story-syntax] [-d|--definitions DEFINITIONS] [--snippets-for [SNIPPETS-FOR]] [--snippets-type SNIPPETS-TYPE] [--append-snippets] [--no-snippets] [--strict] [--order ORDER] [--rerun] [--stop-on-failure] [--dry-run] [--] [<paths>]


Process finished with exit code 1
Empty test suite.
```

Reply:
Can you provide a backtrace so we can see which code is passing a relative path? Try running the test with the `-vvv` option.

Reply:
As an extra help for this case, when you apply this patch, you should also delete the tmp cache that behat creates. For me, applying the patch was also causing this exception to throw in PhpStorm.
Running `rm -rf /tmp/behat_*` solved it.

Reply:
Resolving this conversation, since no reply was given and it is probably just working as expected. The exception is thrown when a relative path is passed, which is wrong and no longer allowed thanks to this PR. The bug is probably in the calling code.